src/pretix/presale/ical.py

```diff
@@ -122,61 +124,110 @@ def get_private_icals(event, positions):
 
     creation_time = datetime.datetime.now(datetime.timezone.utc)
     calobjects = []
-
-    evs = set(p.subevent or event for p in positions)
-    for ev in evs:
-        if isinstance(ev, Event):
+    calentries = set()  # using set for automatic deduplication of CalEntries
+    CalEntry = namedtuple('CalEntry', ['summary', 'description', 'location', 'dtstart', 'dtend', 'uid'])
+
+    # collecting the positions' calendar entries, preferring the most exact date and time available (positions > subevent > event)
+    prefetch_related_objects(positions, 'item__program_times')
+    for p in positions:
+        program_times = p.item.program_times.all()
+        if program_times:
+            # if program times have been configured, they are preferred for the position's calendar entries
             url = build_absolute_uri(event, 'presale:event.index')
+            for index, pt in enumerate(program_times):
+                summary = _('{event} - {item}').format(event=event, item=p.item.name)
```

Comment:
I think we want use the subevent if available?

```suggestion
                summary = _('{event} - {item}').format(event=p.subevent or event, item=p.item.name)
```